packages/core/src/__integrationtests__/PublicCredentials.spec.ts

```diff
@@ -0,0 +1,315 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @group integration/publicCredentials
+ */
+
+import type { ApiPromise } from '@polkadot/api'
+import type {
+  AssetDidUri,
+  DidDocument,
+  IPublicCredentialInput,
+  KiltKeyringPair,
+} from '@kiltprotocol/types'
+
+import { randomAsHex } from '@polkadot/util-crypto'
+import * as Did from '@kiltprotocol/did'
+import {
+  createFullDidFromSeed,
+  KeyTool,
+  makeSigningKeyTool,
+} from '@kiltprotocol/testing'
+import { UUID } from '@kiltprotocol/utils'
+import * as CType from '../ctype'
+import * as PublicCredential from '../publicCredential'
+import {
+  createEndowedTestAccount,
+  initializeApi,
+  isCtypeOnChain,
+  nftNameCType,
+  submitTx,
+} from './utils'
+import { disconnect } from '../kilt'
+
+let tokenHolder: KiltKeyringPair
+let attester: DidDocument
+let attesterKey: KeyTool
+
+let api: ApiPromise
+// Generate a random asset ID
+const assetId: AssetDidUri = `did:asset:eip155:1.erc20:${randomAsHex(20)}`
+let latestCredential: IPublicCredentialInput
+
+beforeAll(async () => {
+  api = await initializeApi()
+}, 30_000)
+
+beforeAll(async () => {
+  tokenHolder = await createEndowedTestAccount()
+  attesterKey = makeSigningKeyTool()
+  attester = await createFullDidFromSeed(tokenHolder, attesterKey.keypair)
+}, 60_000)
+
+describe('When there is an attester and ctype NFT name', () => {
+  beforeAll(async () => {
+    const ctypeExists = await isCtypeOnChain(nftNameCType)
+    if (ctypeExists) return
+    const tx = await Did.authorizeTx(
+      attester.uri,
+      api.tx.ctype.add(CType.toChain(nftNameCType)),
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(tx, tokenHolder)
+  }, 60_000)
+
+  it('should be possible to issue a credential', async () => {
+    latestCredential = {
+      claims: {
+        name: `Certified NFT collection with id ${UUID.generate()}`,
+      },
+      cTypeHash: CType.getHashForSchema(nftNameCType),
+      delegationId: null,
+      subject: assetId,
+    }
+    const encodedPublicCredential = PublicCredential.toChain(latestCredential)
+    const storeTx = api.tx.publicCredentials.add(encodedPublicCredential)
+    const authorizedStoreTx = await Did.authorizeTx(
+      attester.uri,
```

Comment:
Will the naming become simpler if we inline the `storeTx` constant? The same applies to more transactions in the tests below.